src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2264,8 +2299,14 @@ private bool UseUpdatedEscapeRulesForInvocation(Symbol symbol)
                     return true;
                 }
 
+                // track the widest scope that arguments could safely escape to.
+                // use this scope as the inferred STE of declaration expressions.
+                var inferredDestinationValEscape = CallingMethodScope;
                 foreach (var (parameter, argument, _) in escapeArguments)
                 {
+                    // in the old rules, we assume that refs cannot escape into ref struct variables.
+                    // e.g. in `dest = M(ref arg)`, we assume `ref arg` will not escape into `dest`, but `arg` might.
+                    inferredDestinationValEscape = Math.Max(inferredDestinationValEscape, GetValEscape(argument, scopeOfTheContainingExpression));
                     var valid = CheckValEscape(argument.Syntax, argument, scopeOfTheContainingExpression, escapeTo, false, diagnostics);
 
```

Comment:
>receiver: null, // receiver handled explicitly below [](http://example.com/codeflow?start=16&length=52)

Note: We also pass `receiver: null` in `GetInvocationEscapeScope()` and `CheckInvocationEscape()`. I think it's reasonable to keep those callers as is for now, but mentioning it in case I've overlooked something.